linesearch.go

```diff
@@ -10,132 +10,130 @@ import (
 	"github.com/gonum/floats"
 )
 
-// Linesearch is a linesearch-based optimization method.
-// It consists of a NextDirectioner, which specifies the next linesearch method,
-// and a LinesearchMethod which performs the linesearch in the direction specified
-// by the NextDirectioner.
-type Linesearch struct {
+// LinesearchMethod represents an abstract optimization method in which
+// a function is optimized through successive line search optimizations.
+// It consists of a NextDirectioner, which specifies the search direction
+// at each iteration, and a Linesearcher which performs a linesearch along
```

Comment:
I don't like the overloading of the word "iteration". How about "specifies the search direction of each line search, and a Linesearcher which performs a line search along that direction"